elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -507,6 +597,112 @@ defmodule API.Client.Channel do
+               preload: :group
+             ),
+           {:ok, gateways} <-
+             filter_compatible_gateways(gateways, socket.assigns.gateway_version_requirement),
+           OpenTelemetry.Tracer.set_attribute(:gateways_count, length(gateways)),
+           gateway = Gateways.load_balance_gateways(location, gateways, connected_gateway_ids),
+           OpenTelemetry.Tracer.set_attribute(:gateway_id, gateway.id),
+           {:ok, resource, flow} <-
+             Flows.authorize_flow(
+               socket.assigns.client,
+               gateway,
+               resource_id,
+               socket.assigns.subject
+             ) do
+        OpenTelemetry.Tracer.set_attribute(:flow_id, flow.id)
+        opentelemetry_ctx = OpenTelemetry.Ctx.get_current()
+        opentelemetry_span_ctx = OpenTelemetry.Tracer.current_span_ctx()
+
+        preshared_key = generate_preshared_key()
+        ice_credentials = generate_ice_credentials(socket.assigns.client, gateway)
+
+        :ok =
+          Gateways.broadcast_to_gateway(
+            gateway,
+            {:authorize_flow, {self(), socket_ref(socket)},
+             %{
+               client_id: socket.assigns.client.id,
+               resource_id: resource.id,
+               flow_id: flow.id,
+               authorization_expires_at: flow.expires_at,
+               ice_credentials: ice_credentials,
+               preshared_key: preshared_key
+             }, {opentelemetry_ctx, opentelemetry_span_ctx}}
+          )
+
+        {:noreply, socket}
+      else
+        {:error, :not_found} ->
+          OpenTelemetry.Tracer.set_status(:error, "not_found")
+
+          # We are pushing a message instead of replying for the sake of connlib message parsing convenience
+          # {:reply, {:error, %{reason: :not_found}}, socket}
+
+          push(socket, "flow_creation_failed", %{
+            resource_id: resource_id,
+            reason: :not_found
+          })
+
+          {:noreply, socket}
+
+        {:ok, []} ->
+          OpenTelemetry.Tracer.set_status(:error, "offline")
+
+          # We are pushing a message instead of replying for the sake of connlib message parsing convenience
+          # {:reply, {:error, %{reason: :offline}}, socket}
+
+          push(socket, "flow_creation_failed", %{
+            resource_id: resource_id,
+            reason: :offline
+          })
+
+          {:noreply, socket}
+
+        {:error, {:forbidden, violated_properties: violated_properties}} ->
+          OpenTelemetry.Tracer.set_status(:error, "forbidden")
+
+          # We are pushing a message instead of replying for the sake of connlib message parsing convenience
+          # {:reply, {:error, %{reason: :forbidden, violated_properties: violated_properties}}, socket}
+
+          push(socket, "flow_creation_failed", %{
+            resource_id: resource_id,
+            reason: :forbidden,
+            violated_properties: violated_properties
```

Comment:
What is the schema of `violated_properties`?

Reply:
@thomaseizinger it did not change: it's a list of strings:
```
remote_ip_location_region
remote_ip
provider_id
current_utc_datetime
client_verified
```